### ISSUE TITLE
Improve the docstrings

### ISSUE THREAD

**Issue by maximlt** [holoviz/param#588]
I'd like the docstrings of the public API methods and functions of Param to be improved which should improve:
* the API reference page of the website
* the user experience in a code editor

I'm opening this issue to report (later) ideas and findings. For now I would just like to note that Param doesn't seem to follow [PEP257](https://www.python.org/dev/peps/pep-0257/) and also doesn't follow any format (presumably because Param predates the creation of these formats) such as Numpydoc, Google... I quite like [Google's style](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) which is less verbose than Numpy's one.

**Comment by jbednar** [comment]
Right; Param predates just about everything good in Python, and its docstrings are a historical record of whatever the trends may have prevailed at a given time over the past couple of decades. I'm happy with Google style.
